airbyte-webapp/src/components/StatusIcon/StatusIcon.tsx

```diff
@@ -44,8 +46,8 @@ const Container = styled.div<StatusIconProps>`
   vertical-align: middle;
 `;
 
-const Badge = styled(Container)<StatusIconProps>`
-  background: ${(props) => props.theme[(props.status && _themeByStatus[props.status]) || "dangerColor"]};
+const Badge = styled(Container)<{ status: Exclude<StatusIconStatus, "loading"> }>`
```

Comment:
:information_source: Just require the properties that are actually used here (and in the `Pick` above), can simplify that code, since it understands that `theme[...]` cannot be undefined anymore.